src/integrationtests/java/com/aws/greengrass/integrationtests/ShadowManagerTest.java

```diff
@@ -247,4 +264,51 @@ void GIVEN_shadow_manager_WHEN_strategy_config_resets_THEN_respond_to_config_upd
         kernel.getContext().waitForPublishQueueToClear();
         assertThat(syncHandler.getOverallSyncStrategy(), instanceOf(PeriodicSyncStrategy.class));
     }
+
+    @Test
+    void GIVEN_cryptoKeyProviderService_WHEN_get_client_THEN_wait_for_cryptoservice(ExtensionContext context) throws InterruptedException, TLSAuthException, IoTDataPlaneClientCreationException {
+        ignoreExceptionOfType(context, TLSAuthException.class);
+        ignoreExceptionOfType(context, RetryableException.class);
+        ignoreExceptionOfType(context, ResourceNotFoundException.class);
+
+        IotDataPlaneClientFactory factory = spy(kernel.getContext().get(IotDataPlaneClientFactory.class));
+        kernel.getContext().put(IotDataPlaneClientFactory.class, factory);
+        IotDataPlaneClientWrapper wrapper = spy(new FakeIotDataPlaneClientWrapper(factory));
+        kernel.getContext().put(IotDataPlaneClientWrapper.class, wrapper);
+
+        startNucleusWithConfig(NucleusLaunchUtilsConfig.builder()
+                .configFile("sync.yaml")
+                .mqttConnected(true)
+                .mockCloud(false)
+                .build());
+        BaseSyncStrategy s = kernel.getContext().get(RealTimeSyncStrategy.class);
+        assertThat("syncing has started", s::isSyncing, eventuallyEval(is(true)));
+        verify(wrapper, timeout(5000).atLeast(1)).getThingShadow("Thing1", "");
+
+        CountDownLatch cdl=new CountDownLatch(1);
+        SecurityService ss= mock(SecurityService.class);
+        when(ss.getDeviceIdentityKeyManagers()).thenAnswer((invocation)->{
+            cdl.countDown();
+            return new KeyManager[0];
+        });
+        kernel.getContext().put(SecurityService.class, ss);
+        assertThat("request is retried with a new client",cdl.await(10, TimeUnit.SECONDS), is(true));
+        verify(wrapper, timeout(5000).atLeast(2)).getThingShadow("Thing1", "");
+    }
+
+    static class FakeIotDataPlaneClientWrapper extends  IotDataPlaneClientWrapper{
```

Comment:
Let's move this into its own file, so it is more easily usable by other test cases in the future

```suggestion
    static class FakeIotDataPlaneClientWrapper extends IotDataPlaneClientWrapper {
```